File: src/main/java/no/nav/pto_schema/kafka/json/topic/onprem/EndringPaNyForVeilederV1.java
package no.nav.pto_schema.kafka.json.topic.onprem;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Value;

@Value
@NoArgsConstructor(force = true)
@AllArgsConstructor
@Builder
public class EndringPaNyForVeilederV1 {
    String aktorId;
    boolean nyForVeileder;
}

File: src/main/kotlin/no/nav/pto_schema/kafka/json/topic/onprem/EndringPaNyForVeilederV1.kt
package no.nav.pto_schema.kafka.json.topic.onprem

data class EndringPaNyForVeilederV1(
    val aktorId: String,
    val nyForVeileder: Boolean,
) {
    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var aktorId: String? = null
        private var nyForVeileder: Boolean = false

        fun aktorId(aktorId: String?): Builder = apply { this.aktorId = aktorId }
        fun nyForVeileder(nyForVeileder: Boolean): Builder = apply { this.nyForVeileder = nyForVeileder }

        fun build(): EndringPaNyForVeilederV1 = EndringPaNyForVeilederV1(aktorId!!, nyForVeileder)
    }
}

